File: backend/GameApp.Application/Services/SceneServices/SceneGetService.cs
using GameApp.Application.Enumerates;
using GameApp.Application.UseCases.SceneUseCases;
using GameApp.Domain.Entities.Scenes;
using GameApp.Domain.Repositories;
using GameApp.Domain.ValueObjects.Scenes;

namespace SceneApp.Application.Services.SceneServices;

public class SceneGetService : SceneGetUseCase
{
    private ISceneRepository _repo;

    public SceneGetService(ISceneRepository repo)
    {
        _repo = repo;
    }

    public async Task<Scene?> GetScene(Guid id)
    {
        return await _repo.FetchByIdAsync(id);
    }

    public async Task<Scene?> GetSceneByName(SceneName name)
    {
        return await _repo.FetchByName(name);
    }

    public async Task<IEnumerable<Scene>> GetAllScenes()
    {
        return await _repo.FetchAllAsync();
    }

    public async Task<IEnumerable<FinalScene>> GetAllFinalScenes()
    {
        return await _repo.FetchAllFinalScenes();
    }
}

File: frontend/gameapp-frontend/src/application/services/scene-services/scene-get-service.ts
import type { ISceneGetUseCase } from "../../usecases/scene-use-cases/scene-get-use-case";
import type { ISceneRepository } from "../../repositories/iscene-repository";
import { Scene } from "../../../domain/entities/scenes/scene";
import type { SceneName } from "../../../domain/value-objects/scenes/scene-name";
import type { FinalScene } from "../../../domain/entities/scenes/final-scene";

export class SceneGetService implements ISceneGetUseCase {
    constructor(private sceneRepository: ISceneRepository) {}

    async getScene(id: string): Promise<Scene> {
        return await this.sceneRepository.fetchById(id);
    }

    async getSceneByName(name: SceneName): Promise<Scene> {
        return await this.sceneRepository.fetchByName(name);
    }

    async getAllScenes(): Promise<Scene[]> {
        return await this.sceneRepository.fetchAll();
    }

    async getAllFinalScenes(): Promise<FinalScene[]> {
        return await this.sceneRepository.fetchAllFinalScenes();
    }   
}
